### PR TITLE
async workaround

### PR DESCRIPTION
**Pull request by kaskar2008**
Simple async/await workaround for asynchronous route handlers.
In addition: controller context fix.
Node.js version is now must be ^7.10.1.

Consider discuss.

### CODE REVIEW

**Comment by Raiondesu** [comment]
@alexa026 (or @lexa026), please, review the pull-request.

File: src/Ioc/index.js [hunk 1]
```diff
@@ -133,11 +133,15 @@ class Ioc {
      * @param parameters
      * @return {*}
      */
-    static call(binding, functionName, parameters = []) {
+    static async call(binding, functionName, parameters = []) {
```
Comment (alexa026) [inline src/Ioc/index.js:136]: call function cannot be async, as it's used by several different things, a lot of stuff would change if we make it async.

File: src/Router/index.js [hunk 2]
```diff
@@ -148,10 +165,14 @@ class Router {
      * @param response
      * @return {*}
      */
-    static dispatchRoute(route, response) {
+    static async dispatchRoute(route, response) {
```
Comment (alexa026) [inline src/Router/index.js:168]: It doesn't have to be async, it should also allow the response that is NOT promise.

Comment (kaskar2008) [reply src/Router/index.js:168]: async/await construction CAN work with not a promise response.

File: src/Ioc/index.js (hunk 1 above)
Comment (kaskar2008) [reply src/Ioc/index.js:136]: Can you provide me an example? I think I did all the necessary tweaks.

Comment (Raiondesu) [reply src/Ioc/index.js:136]: Async/await can be used without returning promises.
A sentence like:
```js
const result = await someFunction();
```
Assigns value to the result even if `someFunction` doesn't return a promise.
Same as:
```js
const result = someFunction();
```

So, current proposal won't change anything, as `call` function can be used both with and without `await`.

File: src/Router/index.js (hunk 2 above)
Comment (alexa026) [reply src/Router/index.js:168]: It does, but it always returns a Promise, doesn't it? I don't see promise handler added on `respondToRoute` function, have you tested it?

Comment (kaskar2008) [reply src/Router/index.js:168]: `handlerResponse` is getting the response with `await`, so `respondToRoute` can be as is.

Comment (kaskar2008) [reply src/Router/index.js:168]: And, of course I have tested all the fixes.

File: src/Ioc/index.js (hunk 1 above)
Comment (Raiondesu) [reply src/Ioc/index.js:136]: All the tests needed are added to the respected test files, BTW.

**Pull request merged**